Error when clearing percentage of cat & weighting category

When clearing out the percentage for a category when the GB is set to "Categories & Weighting," received the following stacktrace error:

`org.apache.wicket.WicketRuntimeException: Method onRequest of interface org.apache.wicket.behavior.IBehaviorListener targeted at org.sakaiproject.gradebookng.tool.panels.SettingsCategoryPanel$8$3@7a7ff992 on component [TextField [Component id = weight]] threw an exception
	at org.apache.wicket.RequestListenerInterface.internalInvoke(RequestListenerInterface.java:268)
	at org.apache.wicket.RequestListenerInterface.invoke(RequestListenerInterface.java:241)
	at org.apache.wicket.core.request.handler.ListenerInterfaceRequestHandler.invokeListener(ListenerInterfaceRequestHandler.java:250)
	at org.apache.wicket.core.request.handler.ListenerInterfaceRequestHandler.respond(ListenerInterfaceRequestHandler.java:236)
	at org.apache.wicket.request.cycle.RequestCycle$HandlerExecutor.respond(RequestCycle.java:862)
	at org.apache.wicket.request.RequestHandlerStack.execute(RequestHandlerStack.java:64)
	at org.apache.wicket.request.cycle.RequestCycle.execute(RequestCycle.java:261)
	at org.apache.wicket.request.cycle.RequestCycle.processRequest(RequestCycle.java:218)
	at org.apache.wicket.request.cycle.RequestCycle.processRequestAndDetach(RequestCycle.java:289)
	at org.apache.wicket.protocol.http.WicketFilter.processRequestCycle(WicketFilter.java:259)
	at org.apache.wicket.protocol.http.WicketFilter.processRequest(WicketFilter.java:201)
	at org.apache.wicket.protocol.http.WicketServlet.doPost(WicketServlet.java:159)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.sakaiproject.util.RequestFilter.doFilter(RequestFilter.java:410)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:721)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:466)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:358)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:318)
	at org.sakaiproject.tool.impl.ActiveToolComponent$MyActiveTool.forward(ActiveToolComponent.java:513)
	at org.sakaiproject.portal.charon.SkinnableCharonPortal.forwardTool(SkinnableCharonPortal.java:1481)
	at org.sakaiproject.portal.charon.handlers.SiteHandler.doSite(SiteHandler.java:478)
	at org.sakaiproject.portal.charon.handlers.SiteHandler.doGet(SiteHandler.java:271)
	at org.sakaiproject.portal.charon.handlers.SiteHandler.doGet(SiteHandler.java:152)
	at org.sakaiproject.portal.charon.handlers.WorksiteHandler.doPost(WorksiteHandler.java:71)
	at org.sakaiproject.portal.charon.SkinnableCharonPortal.doPost(SkinnableCharonPortal.java:1134)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.sakaiproject.util.RequestFilter.doFilter(RequestFilter.java:471)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:142)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:617)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:518)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1091)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:668)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1521)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1478)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor818.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.apache.wicket.RequestListenerInterface.internalInvoke(RequestListenerInterface.java:258)
	... 55 more
Caused by: java.lang.NullPointerException
	at org.sakaiproject.gradebookng.tool.panels.SettingsCategoryPanel.updateRunningTotal(SettingsCategoryPanel.java:565)
	at org.sakaiproject.gradebookng.tool.panels.SettingsCategoryPanel.updateRunningTotal(SettingsCategoryPanel.java:596)
	at org.sakaiproject.gradebookng.tool.panels.SettingsCategoryPanel.access$100(SettingsCategoryPanel.java:46)
	at org.sakaiproject.gradebookng.tool.panels.SettingsCategoryPanel$8$3.onUpdate(SettingsCategoryPanel.java:317)
	at org.apache.wicket.ajax.form.AjaxFormComponentUpdatingBehavior.onEvent(AjaxFormComponentUpdatingBehavior.java:153)
	at org.apache.wicket.ajax.AjaxEventBehavior.respond(AjaxEventBehavior.java:124)
	at org.apache.wicket.ajax.AbstractDefaultAjaxBehavior.onRequest(AbstractDefaultAjaxBehavior.java:633)
	... 59 more`

--- a/gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/util/FormatHelper.java
+++ b/gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/util/FormatHelper.java
@@ -8,6 +8,9 @@ import java.util.Date;
 
 import org.apache.commons.lang.StringUtils;
 
+import lombok.extern.apachecommons.CommonsLog;
+
+@CommonsLog
 public class FormatHelper {
 
 	/**
@@ -63,7 +66,17 @@ public class FormatHelper {
 		if (StringUtils.isBlank(grade)) {
 			return "";
 		}
-		return StringUtils.removeEnd(String.valueOf(Double.parseDouble(grade)), ".0");
+
+		String s = null;
+		try {
+			final Double d = Double.parseDouble(grade);
+			s = String.valueOf(d);
+		} catch (final NumberFormatException e) {
+			log.debug("Bad format, returning original string: " + grade);
+			s = grade;
+		}
+
+		return StringUtils.removeEnd(s, ".0");
 	}
 
 	/**

--- a/gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java
+++ b/gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java
@@ -2,7 +2,9 @@ package org.sakaiproject.gradebookng.tool.panels;
 
 import java.math.BigDecimal;
 import java.math.RoundingMode;
+import java.text.DecimalFormat;
 import java.text.NumberFormat;
+import java.text.ParseException;
 import java.util.Collections;
 import java.util.List;
 import java.util.Locale;
@@ -505,12 +507,6 @@ public class SettingsCategoryPanel extends Panel {
 		@Override
 		public Double convertToObject(final String value, final Locale locale) throws ConversionException {
 
-			// want this truncated to four decimal places, or less
-			final NumberFormat df = NumberFormat.getInstance();
-			df.setMinimumFractionDigits(0);
-			df.setMaximumFractionDigits(4);
-			df.setRoundingMode(RoundingMode.HALF_UP);
-
 			// convert
 			Double d;
 			try {
@@ -519,11 +515,21 @@ public class SettingsCategoryPanel extends Panel {
 				throw new ConversionException(e).setResourceKey("settingspage.update.failure.categoryweightnumber");
 			}
 
-			// to string for the rounding/truncation
+			// want this truncated to four decimal places, or less
+			// format, then parse back into a double
+			final DecimalFormat df = new DecimalFormat();
+			df.setMinimumFractionDigits(0);
+			df.setMaximumFractionDigits(4);
+			df.setRoundingMode(RoundingMode.HALF_UP);
+
 			final String s = df.format(d);
 
-			// back to double
-			return Double.valueOf(s);
+			try {
+				return df.parse(s).doubleValue();
+			} catch (final ParseException e) {
+				throw new ConversionException(e).setResourceKey("settingspage.update.failure.categoryweightnumber");
+			}
+
 		}
 
 		/**
@@ -562,7 +568,12 @@ public class SettingsCategoryPanel extends Panel {
 		BigDecimal total = BigDecimal.ZERO;
 		for (final CategoryDefinition categoryDefinition : categories) {
 
-			BigDecimal weight = BigDecimal.valueOf(categoryDefinition.getWeight());
+			Double catWeight = categoryDefinition.getWeight();
+			if (catWeight == null) {
+				catWeight = 0D;
+			}
+
+			BigDecimal weight = BigDecimal.valueOf(catWeight);
 			if (weight == null) {
 				weight = BigDecimal.ZERO;
 			}

